### ISSUE TITLE
Cannot read property 'c' of undefined in unit test

### ISSUE THREAD

**Issue by ritchieanesco** [AlexxNB/tinro#36]
I'm using [Testing Library](https://testing-library.com/docs/svelte-testing-library/intro/) which uses Jest to simulate the browser experience. Below is example of how I am testing routes. 

```
it("should navigate to hierarchy page", async () => {
  render(App)
  await fireEvent.click(nav.getByText(/hierarchy/i))
  await waitFor(() => screen.getByTestId("hierarchy-page"))
  expect(screen.getByTestId("hierarchy-page")).toBeInTheDocument()

  await fireEvent.click(nav.getByText(/business/i))
  await waitFor(() => screen.getByTestId("business-page"))
  expect(screen.getByTestId("business-page")).toBeInTheDocument()
})
```

The following error happens when i navigate to another page:
`TypeError: Cannot read property 'c' of undefined`.

Node provides some additional information: "UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch()."

The call stack:
```
at C (node_modules/tinro/dist/tinro.js:1:1782)
      at Object.o (node_modules/tinro/dist/tinro.js:1:7283)
      at transition_out (node_modules/svelte/internal/index.js:820:15)
      at Object.o (src/routes/routes.mock.svelte:519:4)
      at transition_out (node_modules/svelte/internal/index.js:820:15)
      at Object.o (src/App.svelte:42:4)
      at transition_out (node_modules/svelte/internal/index.js:820:15)
      at Object.o (src/template/base.svelte:62:4)
      at transition_out (node_modules/svelte/internal/index.js:820:15)
      at Object.p (src/template/base.svelte:113:5)
      at update (node_modules/svelte/internal/index.js:768:36)
      at flush (node_modules/svelte/internal/index.js:736:13)
      at update (node_modules/svelte/internal/index.js:1085:13)
      at node_modules/svelte/internal/index.js:1092:13
 ```
Any idea what is going on?

**Comment by ritchieanesco** [comment]
ok looks like this issue is coming from svelte and seems to be an ongoing thing. https://github.com/sveltejs/svelte/issues/3165.  Is it possible to publish the source files in order to address this issue as suggested by this [comment](https://github.com/sveltejs/svelte/issues/3165#issuecomment-747910620)

**Comment by ritchieanesco** [comment]
For a temporary solution i've created a patch using [patch package](https://www.npmjs.com/package/patch-package). Below is the patch for v0.4.6. This patch simply replaces import statements that reference the distributable folder with source folder.

```
diff --git a/node_modules/tinro/.DS_Store b/node_modules/tinro/.DS_Store
new file mode 100644
index 0000000..646c355
Binary files /dev/null and b/node_modules/tinro/.DS_Store differ
diff --git a/node_modules/tinro/cmp/Route.svelte b/node_modules/tinro/cmp/Route.svelte
index 9a17f54..bb104f1 100644
--- a/node_modules/tinro/cmp/Route.svelte
+++ b/node_modules/tinro/cmp/Route.svelte
@@ -1,5 +1,5 @@
 <script>
-    import {createRouteObject} from './../dist/tinro_lib';
+    import {createRouteObject} from '../src/lib';
 
     export let path = '/*';
     export let fallback = false;
diff --git a/node_modules/tinro/cmp/index.js b/node_modules/tinro/cmp/index.js
index eaa0bb0..6ebb7e9 100644
--- a/node_modules/tinro/cmp/index.js
+++ b/node_modules/tinro/cmp/index.js
@@ -1,2 +1,2 @@
 export {default as Route} from './Route.svelte';
-export {router,active} from './../dist/tinro_lib';
\ No newline at end of file
+export {router,active} from '../src/router';
diff --git a/node_modules/tinro/src/index.js b/node_modules/tinro/src/index.js
new file mode 100644
index 0000000..65ccf50
--- /dev/null
+++ b/node_modules/tinro/src/index.js
@@ -0,0 +1,2 @@
+export {default as Route} from '../cmp/Route.svelte';
+export * from './tinro';
diff --git a/node_modules/tinro/src/lib.js b/node_modules/tinro/src/lib.js
new file mode 100644
index 0000000..e2352b3
--- /dev/null
+++ b/node_modules/tinro/src/lib.js
@@ -0,0 +1,166 @@
+import {getContext,setContext,onMount,tick} from 'svelte';
+import {router} from './router';
+
+export function createRouteObject(options){
+
+    const parent = getContext('tinro');
+
+    if(parent && (parent.exact || parent.fallback) ) err(
+        `${options.fallback ? '<Route fallback>' : `<Route path="${options.path}">`}  can't be inside ${parent.fallback ? 
+            '<Route fallback>' :
+            `<Route path="${parent.path || '/'}"> with exact path` }`
+    );
+
+    const type = options.fallback ? 'fallbacks' : 'childs';
+
+    const route = {
+        un:null,
+        exact: false,
+        pattern: '',
+        params: {},
+        parent,
+        fallback: options.fallback,
+        redirect: options.redirect,
+        childs: new Set(),
+        activeChilds: new Set(),
+        fallbacks: new Set(),
+        makePattern(path){
+            route.exact = !path.endsWith('/*');
+            route.pattern = formatPath(`${route.parent && route.parent.pattern || ''}${path}`)
+        },
+        register: () => {
+            if(!route.parent) return;
+            route.parent[type].add(route);
+            return ()=>{
+                route.parent[type].delete(route);
+                route.un && route.un();
+            }
+        },
+        show: ()=>{
+            options.onShow();
+            !route.fallback && route.parent && route.parent.activeChilds.add(route);
+        },
+        hide: ()=>{
+            options.onHide();
+            !route.fallback && route.parent && route.parent.activeChilds.delete(route);
+        },
+        match: async (url)=>{
+            const params = getRouteData(route.pattern,url);
+
+            if(params && route.redirect && (!route.exact || (route.exact && params.exact))){
+                return router.goto(makeRedirectURL(url,route.parent.pattern,route.redirect));
+            }
+
+            if(!route.fallback && params && (!route.exact || (route.exact && params.exact))){
+                options.onParams(route.params = params.params);
+                route.show();
+            }else{
+                route.hide();
+            }
+
+            await tick();
+          
+            if(
+                !route.fallback && params && 
+                (
+                    (route.childs.size > 0 && route.activeChilds.size == 0) ||
+                    (route.childs.size == 0 && route.fallbacks.size > 0)
+                )
+            ){
+                let obj = route;
+                while(obj.fallbacks.size == 0){
+                    obj = obj.parent;
+                    if(!obj) return;
+                }
+                obj && obj.fallbacks.forEach(fb => fb.show());
+            }
+        }
+    }
+
+    route.makePattern(options.path);
+
+    setContext('tinro',route);
+    onMount(()=>route.register());
+
+    route.un = router.subscribe(r => {
+        route.match(r.path);
+    });
+    
+    return route;
+}
+
+
+export function formatPath(path,slash=false){
+    path = path.slice(
+        path.startsWith('/#') ? 2 : 0,
+        path.endsWith('/*') ? -2 : undefined
+    )
+    if(!path.startsWith('/')) path = '/'+path;
+    if(path==='/') path = '';
+    if(slash && !path.endsWith('/')) path += '/';
+    return path;
+}
+
+export function getRouteData(pattern,path){
+    pattern = formatPath(pattern,true);
+    path = formatPath(path,true);
+
+    const keys = [];
+    let params = {};
+    let exact = true;
+    let rx = pattern
+       .split('/')
+       .map(s => s.startsWith(':') ? (keys.push(s.slice(1)),'([^\\/]+)') : s)
+       .join('\\/');
+
+    let match = path.match(new RegExp(`^${rx}$`));
+    if(!match) {
+        exact = false;
+        match = path.match(new RegExp(`^${rx}`));
+    }
+    if(!match) return null;
+    keys.forEach((key,i) => params[key] = match[i+1]);
+
+    return {exact,params};
+}
+
+export function makeRedirectURL(path,parent_pattern,slug){
+    if(slug === '') return path;
+    if(slug[0] === '/') return slug;
+    const getParts = url => url.split('/').filter(p=>p!=='');
+
+    const pathParts = getParts(path);
+    const patternParts = getParts(parent_pattern);
+
+    return '/'+patternParts.map((_,i)=>pathParts[i]).join('/')+'/'+slug;
+}
+
+export function getAttr(node,attr,rm,def){
+    const re = [attr,'data-'+attr].reduce( 
+        (r,c) => {
+            const a = node.getAttribute(c);
+            if(rm) node.removeAttribute(c);
+            return a === null ? r: a;
+        },
+    false );
+    return !def && re === '' ? true : re ? re : def ? def : false;
+}
+
+export function parseQuery(str){
+    const o= str.split('&')
+      .map(p => p.split('='))
+      .reduce((r,p) => {
+          const name = p[0];
+          if(!name) return r;
+          let value = p.length > 1 ? p[p.length-1] : true;
+          if(typeof value === 'string' && value.includes(',')) value = value.split(',');
+          (r[name] === undefined) ? r[name]=[value] : r[name].push(value);
+          return r;
+      },{});
+  
+    return Object.entries(o).reduce((r,p)=>(r[p[0]]=p[1].length>1 ? p[1] : p[1][0],r),{});
+}
+
+function err(text){
+    throw new Error(text);
+}
\ No newline at end of file
diff --git a/node_modules/tinro/src/router.js b/node_modules/tinro/src/router.js
new file mode 100644
index 0000000..ec437b3
--- /dev/null
+++ b/node_modules/tinro/src/router.js
@@ -0,0 +1,81 @@
+import {getContext} from 'svelte';
+import {writable} from 'svelte/store';
+import {getAttr,parseQuery,getRouteData} from './lib';
+
+export const router = routerStore();
+
+export function active(node){
+    const href = getAttr(node,'href'),
+          exact = getAttr(node,'exact',true),
+          cl = getAttr(node,'active-class',true,'active');
+          
+    return {destroy:router.subscribe(r => {
+        const data = getRouteData(href,r.path); 
+        data && (data.exact && exact || !exact) ? node.classList.add(cl) : node.classList.remove(cl);
+    })}
+}
+
+function routerStore(){
+    let hsh = window.location.pathname === 'srcdoc';
+
+    const go = (href,set) => {
+        hsh ? window.location.hash=href : history.pushState({}, '', href);
+        set(getLocation(hsh));
+    }
+
+    const {subscribe,set} = writable(getLocation(hsh), _ => {
+        window.hashchange = window.onpopstate = _ => set(getLocation(hsh));
+        const un = aClickListener(href=>go(href,set));
+        return _ => {
+            window.hashchange = window.onpopstate = null;
+            un();
+        }
+    }); 
+
+    return {
+        subscribe,
+        goto: href => go(href,set),
+        params: getParams,
+        useHashNavigation: s => set(getLocation(hsh = s===undefined ? true : s))
+    }
+}
+
+function getLocation(hsh){
+    return hsh ? getLocationFromHash() : {
+        path: window.location.pathname,
+        query: parseQuery(window.location.search.slice(1)),
+        hash: window.location.hash.slice(1)
+    }
+}
+
+function getLocationFromHash(){
+    const match = String(window.location.hash.slice(1)||'/').match(/^([^?#]+)(?:\?([^#]+))?(?:\#(.+))?$/);  
+    return {
+      path: match[1] || '',
+      query: parseQuery(match[2] || ''),
+      hash: match[3] || '',
+    };
+ }
+
+function aClickListener(go){
+    const h = e => {
+        const a = e.target.closest('a[href]');
+        const i = a && getAttr(a,'tinro-ignore');
+
+        if(!i && a){
+            const href = a.getAttribute('href').replace(/^\/#/,'');
+
+            if(!/^\/\/|^[a-zA-Z]+:/.test(href)) {
+                e.preventDefault();
+                go(href.startsWith('/') ? href : a.href.replace(window.location.origin,''));
+            }
+        }
+    }
+
+    addEventListener('click', h);
+    return () => removeEventListener('click', h);
+}
+
+function getParams(){
+    return getContext('tinro').params;
+}
diff --git a/node_modules/tinro/src/tinro.js b/node_modules/tinro/src/tinro.js
new file mode 100644
index 0000000..d840858
--- /dev/null
+++ b/node_modules/tinro/src/tinro.js
@@ -0,0 +1,5 @@
+import {router,active} from './router.js';
+import {createRouteObject} from './lib.js';
+//import {routes,registerRouteObject} from './routes.js';
+
+export {router,active,createRouteObject};
\ No newline at end of file

```

**Comment by ritchieanesco** [comment]
@AlexxNB this issue seems to be resolved with the latest update (0.5.3).

**Comment by AlexxNB** [comment]
I moved bundling from Rollup to ESBuild. I think it is why module working great now.
